src/evaluate/evaluator/question_answering.py

```diff
@@ -0,0 +1,191 @@
+            confidence_level (`float`, defaults to `0.95`):
+                The `confidence_level` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            n_resamples (`int`, defaults to `9999`):
+                The `n_resamples` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            random_state (`int`, *optional*, defaults to `None`):
+                The `random_state` value passed to `bootstrap` if `"bootstrap"` strategy is chosen. Useful for
+                debugging.
+            question_column (`str`, defaults to `"question"`):
+                the name of the column containing the question in the dataset specified by `data`.
+            context_column (`str`, defaults to `"context"`):
+                the name of the column containing the context in the dataset specified by `data`.
+            id_column (`str`, defaults to `"id"`):
+                the name of the column cointaing the identification field of the question and answer pair in the
+                dataset specified by `data`.
+            label_column (`str`, defaults to `"answers"`):
+                the name of the column containing the answers in the dataset specified by `data`.
+        Return:
+            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
+            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
+            containing the score, the confidence interval and the standard error calculated for each metric key.
+        Examples:
+        ```python
+        >>> from evaluate import evaluator
+        >>> from datasets import Dataset, load_dataset
+        >>> e = evaluator("question-answering")
+        >>> data = load_dataset("squad", split="validation[:2]")
+        >>> results = e.compute(
+        >>>     model_or_pipeline="mrm8488/bert-tiny-finetuned-squadv2",
+        >>>     data=data,
+        >>>     metric="squad",
+        >>>     question_column="question",
+        >>>     context_column="context",
+        >>>     label_column="answers",
+        >>>     strategy="bootstrap",
+        >>>     n_resamples=10,
+        >>>     random_state=0
+        >>> )
+        ```"""
+        data = self.prepare_data(
+            data=data,
+            question_column=question_column,
+            context_column=context_column,
+            id_column=id_column,
+            label_column=label_column,
+        )
+
+        pipe = self.prepare_pipeline(model_or_pipeline=model_or_pipeline, preprocessor=tokenizer)
+
+        metric = self.prepare_metric(metric)
+
+        # Compute predictions
+        predictions = pipe(question=data[question_column], context=data[context_column], padding="max_length")
+        predictions = [
+            {"prediction_text": predictions[i]["answer"], "id": data[i][id_column]} for i in range(len(predictions))
+        ]
+
```

Comment:
Depending on how the review of https://github.com/huggingface/evaluate/pull/178 and which PR lands first we may want to add time metrics here (I'll update it in my PR if this one lands first)